src/typeChecker/typeErrorChecker.ts

```diff
@@ -397,8 +456,21 @@ function addTypeDeclarationsToEnvironment(
         setDeclKind(id.name, node.kind, env)
         break
       case 'TSTypeAliasDeclaration':
+        const alias = node.id.name
+        if (context.chapter >= 2 && (alias === 'Pair' || alias === 'List')) {
+          throw new TypecheckError(
+            node,
+            `Type '${alias}' is a reserved type and cannot be redeclared`
+          )
+        }
```

Comment:
After looking at this, I realised that I can do things like:
```ts
type string = number;
```

For reference, TypeScript disallows this:
```
TS2457: Type alias name cannot be 'string'.
```

Reply:
Got it, will follow TS and disallow overwriting of basic types using type aliases.